### PR TITLE
Rename ssh to script, explicitly control cockpit integration availability

### PR DESCRIPTION
**Pull request by adamruzicka**
Foreman-side counterpart: https://github.com/theforeman/foreman_remote_execution/pull/709

### CODE REVIEW

File: lib/smart_proxy_remote_execution_ssh.rb [hunk 1]
```diff
@@ -62,6 +49,24 @@ def validate_mqtt_settings!
...
       end
 
+      def validate_ssh_settings!
+        return unless requires_configured_ssh?
+        unless private_key_file
+          raise "settings for `ssh_identity_key` not set"
+        end
+
+        unless File.exist?(private_key_file)
+          raise "Ssh public key file #{private_key_file} doesn't exist.\n"\
```
Comment (ehelms) [inline lib/smart_proxy_remote_execution_ssh.rb:59]:
```suggestion
          raise "SSH public key file #{private_key_file} doesn't exist.\n"\
```

File: lib/smart_proxy_remote_execution_ssh.rb [hunk 2]
```diff
@@ -62,6 +49,24 @@ def validate_mqtt_settings!
...
+            "You can generate one with `ssh-keygen -t rsa -b 4096 -f #{private_key_file} -N ''`"
+        end
+
+        unless File.exist?(public_key_file)
+          raise "Ssh public key file #{public_key_file} doesn't exist"
```
Comment (ehelms) [inline lib/smart_proxy_remote_execution_ssh.rb:64]:
```suggestion
          raise "SSH public key file #{public_key_file} doesn't exist"
```

**Comment by ehelms** [comment]
I opened a corresponding puppet PR: https://github.com/theforeman/puppet-foreman_proxy/pull/735

**Review by ehelms — APPROVED** [review]
General :+1: from me

**Review by ezr-ondrej — APPROVED** [review]
Looks good to me :+1:
I don't want to mess up the order as I'm bit lost already, so not merging, but it is ready to go if it is not waiting on anything.

**Comment by adamruzicka** [comment]
REX 6.2.0 was just released, let's get this in.

**Comment by adamruzicka** [comment]
Thank you @ehelms & @ezr-ondrej !

**Pull request merged**